### ISSUE TITLE
Variable Input/Output Transformation advancements are unobtainable

### ISSUE THREAD

**Issue by yoshiweegee** [CyclopsMC/IntegratedDynamics#231]
For some reason, these advancements are not working. I have checked that I have the respective prerequisites. This is problematic because it prevents the player from getting the rewards for the Network Foundations tutorial.

Up to the point where this occurred, I was switching worlds and gamemodes. This could be a lead for why this is happening.

Restarting the game does not fix this.

**Comment by rubensworks** [comment]
MC/mod version?

**Comment by yoshiweegee** [comment]
Apologies, the Minecraft version is 1.12, and the Integrated Dynamics version is 1.12-0.8.3.

Other relevant version information:
- Minecraft Forge: 14.21.1.2443 w/ Mercurius
- CyclopsCore: 1.12-0.10.11
- CommonCapabilities: 1.12-1.3.4

(From the in-game mods list)

**Comment by rubensworks** [comment]
In what kind of crafting table are you crafting the items? Perhaps another mod is preventing crafted events.

**Comment by yoshiweegee** [comment]
I have no other mods except the prerequisites.

**Comment by rubensworks** [comment]
Ok, then it must be something on my end, will look into it.

**Comment by neoc666** [comment]
This bug might have reappeared: I'm playing FTB Revelation for 1.12 in the newest version, added Integrated Dynamics and Tunnels manually. Crafted in Tinker's Crafting Station. First quest gives rewards, second (the one described in this bug) doesn't. I have a lot of mods from the pack disabled, but I doubt that makes a difference.

IntegratedDynamics-1.12.2-0.11.10
IntegratedTunnels-1.12.2-1.5.5

**Comment by rubensworks** [comment]
@neoc666 What about the vanilla crafting table?

**Comment by neoc666** [comment]
I don't think it is related to the crafting table. But I tracked it down a bit more: I had reproducible crashes when trying to do some advanced tutorial (I'll file a crash report later). That forced me to restart my client. After that I noticed that there were green checkmarks on those parts of the tutorial input parts I already had done. Then I noticed that one of the two logic cables in the above mentioned tutorial didn't need just a cable but building a network of 10 cables. So I did that and got the achievement (I always got the achievement, independent of completing the tutorial).

But the tutorial still didn't complete. So I reloaded the world. Nothing changed. But after restarting the client I had completed the tutorial and got my reward. So I guess the mod checks the achievements on client start and if you have completed them, it completes the tutorials. But the tutorials often don't complete immediately after getting the achievement.

Also there are tutorials which don't use a crafting table (logic programmer tutorials). Those had the same issue, so it is hard to imagine the crafting table is relevant here.

FTB Revelation contains BetterAdvancements-1.12.2-0.0.5.17, I don't know if that could mess with events.
